tests/test_cases/test_cocotb/test_edge_triggers.py

```diff
@@ -233,3 +233,39 @@ async def wait_falling_edge():
     falling_coro = cocotb.fork(wait_falling_edge())
     cocotb.fork(Clock(dut.clk, 10, units='ns').start())
     await Combine(rising_coro, falling_coro)
+
+
+@cocotb.test()
+async def test_edge_on_vector(dut):
+    """Test that Edge() triggers on any 0/1 change in a vector"""
+
+    cocotb.fork(Clock(dut.clk, 100, "ns").start())
+
+    edge_cnt = 0
+    edge_cnt_exp = 0
+
+    async def wait_edge():
+        nonlocal edge_cnt
+        while True:
+            await Edge(dut.stream_out_data_registered)
+            edge_cnt = edge_cnt + 1
+
+    cocotb.fork(wait_edge())
+
+    dut.stream_in_data <= 0
+    await ClockCycles(dut.clk, 1)
+
+    for val in range(1, 2**len(dut.stream_in_data)-1):
+        # produce an edge by setting a value != 0:
+        dut.stream_in_data <= val
+        await ClockCycles(dut.clk, 1)
+        edge_cnt_exp = edge_cnt_exp + 1
```

Comment:
Is there some reason you can't precompute what the expected count will be? I figured it would be `2 * (2**len(dut.stream_in_data))`.

Reply:
It's `2 * ((2**len(dut.stream_in_data)-1)-1)` (508) which takes a while to reason about.

Reply:
It's more about verifying the code is meets our expectations. If there is poor simulator behavior in both loops, the test may pass when it shouldn't (probably unlikely though).

Reply:
Changed.